File: src/_cpp/TowerDefenseBase/support/MenuItemTextBG.cpp
//
//  IslandDefense
//
//  Created by Vladimir Tolmachev on 27.09.14.
//
//
#include "MenuItemTextBG.h"
#include "resources.h"
#include "Log.h"
#include "ml/ImageManager.h"
#include "ml/ScrollMenu.h"
using namespace cocos2d;

int g_menuItemImageWithTextCount(0);

MenuItemTextBG * MenuItemTextBG::create(const std::string & text, cocos2d::Color4F colorBG, cocos2d::Color3B colorText, const ccMenuCallback& callback)
{
	__push_auto("MenuItemTextBG::create");
	MenuItemTextBG * ptr = new MenuItemTextBG;
	if ( ptr && ptr->initWithText(text, colorBG, colorText, callback) )
	{
		ptr->autorelease();
		return ptr;
	}
	CC_SAFE_DELETE(ptr);
	return nullptr;
}

bool MenuItemTextBG::initWithText(const std::string & text, cocos2d::Color4F colorBG, cocos2d::Color3B colorText, const ccMenuCallback& callback)
{
	__push_auto_check("MenuItemTextBG::initWithText");
	m_label = LabelTTF::create( text, "Arial", 16 );
	if ( !m_label )
		return false;
	if( !MenuItemLabel::initWithLabel( m_label, callback ) )
		return false;
	m_label->setColor( colorText );

	m_bg = ImageManager::sprite( kPathSpriteSquare );
	m_bg->setAnchorPoint( Point::ZERO );
	m_bg->setColor( Color3B( colorBG.r * 255, colorBG.g * 255, colorBG.b * 255 ) );
	m_bg->setOpacity( colorBG.a * 255 );
	addChild( m_bg, -1 );

	setText( text );
	return true;
}

void MenuItemTextBG::setBorders(float x, float y)
{
	__push_auto_check("MenuItemTextBG::setBorders");
	m_borders = Point(x, y);
}

void MenuItemTextBG::setText(const std::string & text )
{
	__push_auto_check("MenuItemTextBG::setText");
	m_label->setString( text );
	Size size = m_label->getContentSize();
	m_bg->setScale( size.width, size.height );
}

/*******************************************************************************************/

File: src/_eu/support/MenuItemTextBG.js
/******************************************************************************
 * Copyright 2014-2016 Vladimir Tolmachev
 * Copyright 2016 Visionarity AG
 * Vladimir Tolmachev and Visionarity AG have unlimited commercial
 * licenses for commercial use and customization
 *
 * Author: Vladimir Tolmachev (tolm_vl@hotmail.com)
 * Ported C++ to Javascript: Visionarity AG / Vladimir Tolmachev
 * Project: Island Defense (JS)
 * If you received the code not from the author, please contact us
 ******************************************************************************/

//Define namespace
var EU = EU || {};

EU.MenuItemTextBG = cc.MenuItemLabel.extend({

    /** @type {cc.Color} */ m_bgColor : null,
    /** @type {cc.LabelTTF} */ m_label : null,
    /** @type {cc.Point} */ m_borders : null,
    /** @type {cc.Sprite} */ m_bg : null,
    /** @type {MenuItem} */ _shop : null,

    g_menuItemImageWithTextCount: 0,

    ctor: function(text, colorBG,  colorText,  callback)
    {
        cc.log("MenuItemTextBG.create");
        this._super();

        if ( ! this.initWithText(text, colorBG, colorText, callback) )
        {
            return null;
        }
        //CC_SAFE_DELETE(ptr);
        return this;
    },
    
    initWithText: function(text, colorBG,  colorText,  callback)
    {
        cc.log("MenuItemTextBG.initWithText");
        this.m_label = new cc.LabelTTF( text, "Arial", 16 );
        if ( !this.m_label )
            return false;
        if( !this.initWithLabel( this.m_label, callback, null ) )
            return false;
        this.m_label.setColor( colorText );
    
        this.m_bg = EU.ImageManager.sprite( EU.kPathSpriteSquare );
        this.m_bg.setAnchorPoint( cc.POINT_ZERO );
        this.m_bg.setColor( new cc.Color( colorBG.r * 255, colorBG.g * 255, colorBG.b * 255 ) );
        this.m_bg.setOpacity( colorBG.a * 255 );
        this.addChild( this.m_bg, -1 );
    
        this.setText( text );
        return true;
    },
    
    setBorders: function(x, y)
    {
        cc.log("MenuItemTextBG.setBorders");
        this.m_borders = cc.p(x, y);
    },
    
    setText: function(text )
    {
        cc.log("MenuItemTextBG.setText");
        this.m_label.setString( text );
        var size = this.m_label.getContentSize();
        this.m_bg.setScale( size.width, size.height );
    }

});

